### PR TITLE
Create an example which shows a traversal of the "scene graph" in Makie

### PR DESCRIPTION
**Pull request by asinghvi17**
This is pretty interesting, I adapted the code from the existing tree example and will use it for a talk.  Was also thinking of allowing the Scene graph to be visualized either as a `convert_arguments` to `graphplot`, or a standalone recipe.
hello

### CODE REVIEW

File: docs/examples/scenegraph.jl [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    layout=GraphMakie.Buchheim(),
+    nlabels=newl,
+    nlabels_distance=10,
+    nlabels_fontsize=30,
+    nlabels_align,
+    tangents=((0,-1),(0,-1)),
+    figure = (; resolution = (900, 600)),
+    axis = (limits = (-2.5, 2, -16, 2),)
+)
+hidedecorations!(ax); hidespines!(ax)
```
Comment (hexaeder) [inline docs/examples/scenegraph.jl:88]:
```suggestion
hidedecorations!(a); hidespines!(a)
```

File: docs/examples/scenegraph.jl [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    else
+        self = p[:node_pos][][v]
+        parent = p[:node_pos][][inneighbors(newg, v)[1]]
+        if self[1] < parent[1] # left branch
+            nlabels_align[v] = (:right,:center)
+        end
+    end
+end
+p.nlabels_align = nlabels_align
+fig
```
Comment (hexaeder) [inline docs/examples/scenegraph.jl:113]:
```suggestion
@save_reference fig
```
should enable reference testing for the final output. However you'd need to run the tests locally, this should create a new plot in the `assets` subfolder which then can be committed

**Comment by asinghvi17** [comment]
I just noticed that this PR fell off the radar a bit - have cleaned it up and uploaded a refimage, as well as rebasing on the latest master. 
scenegraph jl-01

**Comment by hexaeder** [comment]
thanks for following up on that! :)

**Pull request merged**